Makefile.am

```diff
@@ -8,7 +8,12 @@ if BUILD_OCTAVE_MOD
 octave_dir = Octave
 endif
 
-SUBDIRS = M4 man doc Win32 src $(octave_dir) examples regtest tests programs
+SUBDIRS = M4 Win32 src examples tests
+
+if FULL_SUITE
+SUBDIRS += man doc $(octave_dir) regtest programs
+endif
+
 DIST_SUBDIRS = M4 man doc Win32 src Octave examples regtest tests programs
```

Comment:
In https://github.com/erikd/libsndfile/blob/master/programs/test-sndfile-metadata-set.py#L46 the test suite from `programs/` references stuff in `examples/`.
There's two ways going forward:

1. Either leave it as is
2. or put an explicit make dependency in `programs/Makefile.am`, such that `make_sine` is built, whether the full-suite is enabled or not.

Which one do you prefer?